fuzz/fuzz_targets/bitcoin/script_asm_roundtrip.rs

```diff
@@ -0,0 +1,44 @@
+
+use honggfuzz::fuzz;
+
+fn do_test(data: &[u8]) {
+    let asm = bitcoin::Script::from_bytes(data).to_asm_string();
+    let parsed = bitcoin::ScriptBuf::parse_asm(&asm).unwrap();
+    assert_eq!(parsed.as_bytes(), data);
```

Comment:
I don't think we want to `unwrap` here, any random data is not going to be parsable asm, right?

I think we want:

```rust
    let asm = bitcoin::Script::from_bytes(data).to_asm_string();

    match bitcoin::ScriptBuf::parse_asm(&asm) {
        Ok(parsed) => assert_eq!(parsed.as_bytes(), data),
        Err(_) => return,
    }
```

Reply:
Just to clarify, any random bytes [0] can be converted to asm using `to_asm_string` and should then be parsable?

[0] `ScriptBuf::from_bytes` does not do any checks, just wraps the vector.

Reply:
@tcharding with the current API, yes. We allow the user to embed any sequence of bytes as a `Script`, which does mean that they can do invalid things like having incomplete pushes or the 0xff "not an opcode" opcode or whatever. **However** this behavior matches the Bitcoin network, where the scriptPubkey field of the transaction outputs is allowed to be any random crap.

When we do script tagging it might be interesting to try to make `scriptSig` and `witnessScript`s tighter, since unlike scriptPubKeys they have to be legal for their containing transaction to be legal.

Reply:
Cool, I got ya, thanks. I'm keen to see script tagging.